### PR TITLE
[F] Add settings and reindexing into dev load

### PR DESCRIPTION
**Pull request by SMaxOwok**
[Delivers #138663937]
[Delivers #138667707]

### CODE REVIEW

File: api/app/services/demonstration/data_loader.rb [hunk 1]
```diff
@@ -5,6 +5,11 @@ module Demonstration
   # Loads demo data into the Manifold installation
   class DataLoader
 
+    ENV_SETTINGS = {
+      general: %w(ga_profile_id ga_tracking_id),
```
Comment (zdavis) [inline api/app/services/demonstration/data_loader.rb]: the arrays should be frozen as well.

File: api/app/services/demonstration/data_loader.rb [hunk 2]
```diff
@@ -70,6 +77,25 @@ def create_pages
       end
     end
 
+    def ensure_settings
+      settings = Settings.instance
+      ENV_SETTINGS.each do |key, values|
+        values.each do |value|
+          env_setting = "MANIFOLD_SETTING_#{key}_#{value}".upcase
+          settings[key][value] = settings[key][value] || ENV[env_setting]
```
Comment (zdavis) [inline api/app/services/demonstration/data_loader.rb]: Use ||= here.

```
irb(main):001:0> a = 1
=> 1
irb(main):002:0> a ||= 2
=> 1
irb(main):003:0> a
=> 1
irb(main):004:0> a = nil
=> nil
irb(main):005:0> a ||= 2
=> 2
irb(main):006:0> a
=> 2
```

**Comment by zdavis** [comment]
Rebased onto development to address failing CI build

File: api/app/services/demonstration/data_loader.rb [hunk 3]
```diff
@@ -70,6 +77,25 @@ def create_pages
       end
     end
 
+    def ensure_settings
+      settings = Settings.instance
+      ENV_SETTINGS.each do |key, values|
+        values.each do |value|
+          env_setting = "MANIFOLD_SETTING_#{key}_#{value}".upcase
+          settings[key][value] = settings[key][value] ||= ENV[env_setting]
```
Comment (zdavis) [inline api/app/services/demonstration/data_loader.rb]: You didn't quite get my meaning here. This line should be:

`settings[key][value] ||= ENV[env_setting]`

File: api/app/services/demonstration/data_loader.rb [hunk 4]
```diff
@@ -70,6 +77,25 @@ def create_pages
...
+          @logger.info("Setting updated: #{key}: #{value}".green)
```
Comment (zdavis) [inline api/app/services/demonstration/data_loader.rb]: This log output occurs whether or not the setting was updated, and it doesn't include any information about what it was set to. Not very useful.

Comment (SMaxOwok) [inline api/app/services/demonstration/data_loader.rb]: I took this out, but I wasn't sure if you wanted it in there and give context based feedback.  I can add it back in that manner if you'd rather.

**Comment by zdavis** [comment]
Could you add a couple simple tests to this that prove:

1. The setting value is overwritten if it's already set
2. The setting value is set from env if it's blank
3. Setting these values works even when the settings object doesn't exist.

**Pull request merged**